### PR TITLE
Use GNUInstallDirs for mapping installation directories

### PR DESCRIPTION
**Pull request by jonringer**
Was iterating on this package for Nixpkgs, and found it difficult to consume the `.cmake` configurations as `get_filename_component(_INSTALL_PREFIX "${LIBEVENT_CMAKE_DIR}/../../.." ABSOLUTE)` was not always true.

This should be a no-op for most libevent package maintainers.

### CODE REVIEW

**Review by azat — DISMISSED** [review]

File: cmake/LibeventConfig.cmake.in [hunk 1]
```diff
@@ -157,7 +156,7 @@ set(LIBEVENT_INCLUDE_DIR ${LIBEVENT_INCLUDE_DIRS})
 if(LIBEVENT_LIBRARIES)
     set(LIBEVENT_LIBRARY ${LIBEVENT_LIBRARIES})
     if(CONFIG_FOR_INSTALL_TREE)
-        message_if_needed(STATUS "Found libevent ${LIBEVENT_VERSION} in ${_INSTALL_PREFIX}")
+      message_if_needed(STATUS "Found libevent ${LIBEVENT_VERSION} in @CMAKE_INSTALL_LIBDIR@")
```
Comment (azat) [inline cmake/LibeventConfig.cmake.in:159]:
```suggestion
        message_if_needed(STATUS "Found libevent ${LIBEVENT_VERSION} in @CMAKE_INSTALL_LIBDIR@")
```

Breaks indent

File: CMakeLists.txt [hunk 2]
```diff
@@ -1539,8 +1539,10 @@ endif()
 # Installation preparation.
 #
 
+include(GNUInstallDirs)
+
 set(EVENT_INSTALL_CMAKE_DIR
-    "${CMAKE_INSTALL_PREFIX}/lib/cmake/libevent")
+  "${CMAKE_INSTALL_LIBDIR}/cmake/libevent")
```
Comment (azat) [inline CMakeLists.txt:1545]:
```suggestion
    "${CMAKE_INSTALL_LIBDIR}/cmake/libevent")
```

Let's preserve the indent

**Review by azat — COMMENTED** [review]
LGTM.

But this breaks the export test - https://github.com/libevent/libevent/runs/4050571079?check_suite_focus=true

Can you please take a look?

**Comment by jonringer** [comment]
The test failures were coming from the tests not being able to find `LibeventConfig.cmake`.

I removed the surrounding code which deletes the file and then regenerates it for the test cases. The tests were able to pass locally.

File: CMakeLists.txt [hunk 3]
```diff
@@ -1546,18 +1546,9 @@ set(EVENT_INSTALL_CMAKE_DIR
 
 export(PACKAGE libevent)
 
-function(gen_package_config forinstall)
-    if(${forinstall})
-        set(CONFIG_FOR_INSTALL_TREE 1)
```
Comment (azat) [inline CMakeLists.txt:1549]: So, `CONFIG_FOR_INSTALL_TREE` can be removed from `cmake/LibeventConfig.cmake.in`?

Comment (jonringer) [reply CMakeLists.txt:1549]: I don't see why not. It's being installed to `${CMAKE_BINARY_DIR}`, similar to the other `.cmake` files

Comment (azat) [reply CMakeLists.txt:1549]: But if we will remove `CONFIG_FOR_INSTALL_TREE` code, then `LIBEVENT_INCLUDE_DIRS` will be incorrect.

https://github.com/libevent/libevent/blob/89505f8513b8590e7a7700d481c20562d0726d54/cmake/LibeventConfig.cmake.in#L113-L147

P.S. Sorry for the delay.

**Comment by tristan957** [comment]
What is the status on this PR? It would be nice to add this to the next release of libevent. I don't mind trying to take over the PR.

**Comment by azat** [comment]
>What is the status on this PR? It would be nice to add this to the next release of libevent. I don't mind trying to take over the PR.

I think you can continue it. You can base you patches on top of this to preserve author, as long as you not completely change the patch. Thanks!

**Comment by jonringer** [comment]
sorry, I keep dropping things due to how many balls I'm trying to juggle. If you would like to take over, please do so!

I'll be happy to review :)

**Comment by azat** [comment]
@tristan957 are you going to finish this?

**Comment by tristan957** [comment]
I have it on my list, but if someone else gets to it before I do even better.

**Comment by azat** [comment]
This is still a problem - https://github.com/libevent/libevent/pull/1221#discussion_r753853195

Though by some reason our test for export does not shows the problem.
This should be investigated.

**Comment by azat** [comment]
Merged in #1397

**Pull request closed without merging**